src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -513,23 +513,57 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
     switch (intrinsic)
     {
 #if defined(TARGET_XARCH)
+
         case NI_VectorT_ConvertToDouble:
+        {
+#ifdef TARGET_AMD64
+            if ((varTypeIsLong(simdBaseType) && IsBaselineVector512IsaSupportedOpportunistically()) ||
+                (simdBaseType == TYP_FLOAT && ((simdSize == 32 && compOpportunisticallyDependsOn(InstructionSet_AVX)) ||
+                                               (simdSize == 64 && IsBaselineVector512IsaSupportedOpportunistically()))))
+            {
+                break;
+            }
+#endif // TARGET_AMD64
+            return nullptr;
+        }
+
+        case NI_VectorT_ConvertToInt32:
+        {
+#ifdef TARGET_XARCH
+            if ((simdSize == 16 && compOpportunisticallyDependsOn(InstructionSet_SSE41)) ||
+                (simdSize == 32 && compOpportunisticallyDependsOn(InstructionSet_AVX)) ||
+                IsBaselineVector512IsaSupportedOpportunistically())
```

Comment:
These last two checks should be implicitly true.

We should never encounter `simdSize == 32` when `AVX` is not supported nor `simdSize == 64` when `Avx512F` is not supported.

Reply:
Same goes for other similar checks in this file.

Reply:
I have made appropriate changes in the file.